runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +110,38 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Removes a node from the layout
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
+        Layout newLayout = builder.removeLayoutServer(endpoint)
+                .removeSequencerServer(endpoint)
+                .removeLogunitServer(endpoint)
+                .removeUnResponsiveServer(endpoint)
+                .setEpoch(currentLayout.getEpoch() + 1)
+                .build();
+
+        // Seal after constructing the layout, so that the system
+        // isn't blocked if the builder throws an exception
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
```

Comment:
ideally I would want sealEpoch to return the new epoch and that epoch is set into the newLayout before we attempt consensus, but we can do this later if this will be too much work.